src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -41,7 +41,7 @@ static constexpr uint32_t sampling_rate = 8;
 static_assert(
   gauge_col_t::max_frame_size % sampling_rate == 0, "Invalid sampling rate");
 
-enum class segment_meta_ix {
+enum class segment_meta_ix : size_t {
```

Comment:
why specify an underlying type here?

Reply:
It's temp code. I didn't pay attention to this enum until the last commit, where I understood that the enum is used as a named index into the tuple-of-iterators. I specified the type as a reminder for myself.

Reply:
btw I'm moving the code to another pr https://github.com/redpanda-data/redpanda/pull/9674 so this one can be reviewed more easily while I work on that one